internals/api/auth.go

```diff
@@ -0,0 +1,249 @@
+package api
+
+import (
+	"encoding/json"
+	"net/http"
+	"time"
+
+	"github.com/secrethub/secrethub-go/internals/api/uuid"
+)
+
+// AuthMethod options
+const (
+	AuthMethodAWSSTS = "aws-sts"
+)
+
+// SessionType options
+const (
+	SessionTypeHMAC SessionType = "hmac"
+)
+
+// Errors
+var (
+	ErrInvalidSessionType  = errAPI.Code("invalid_session_type").StatusError("invalid session type provided for authentication request", http.StatusBadRequest)
+	ErrInvalidPayload      = errAPI.Code("invalid_payload").StatusError("invalid payload provided for authentication request", http.StatusBadRequest)
+	ErrInvalidAuthMethod   = errAPI.Code("invalid_auth_method").StatusError("invalid auth method", http.StatusBadRequest)
+	ErrMissingField        = errAPI.Code("missing_field").StatusErrorPref("request is missing field %s", http.StatusBadRequest)
+	ErrSessionNotFound     = errAPI.Code("session_not_found").StatusError("session could not be found, it might have expired", http.StatusForbidden)
```

Comment:
We don't know whether it does not exist or the TTL in Redis has expired. It is _gone_.

Reply:
I know, but let's look at this from the user's perspective. The most likely case when they see this error is that their session expired. Wording it like the possible expiration is a consequence of the not found error unnecessarily exposes an implementation detail to the user that they don't need to be aware of. Being vague in our language would only confuse them more rather than inform them. 

`session not found or expired` makes kinda sense IMO, another alternative is `no valid session found`. 

Anyway, it's not a super big thing. Proceed as you see fit.